cardano-db-sync/src/Cardano/DbSync/Cache.hs

```diff
@@ -95,6 +98,11 @@ data CacheStatistics = CacheStatistics
   , prevBlockQueries :: !Word64
   }
 
+data CacheEpoch = CacheEpoch
+  { ceEpoch :: !(Maybe DB.Epoch)
```

Comment:
Is it ever the case where this is Nothing and BlockId is Just? If not then we could change the type to `Maybe (CacheEpoch DB.Epoch (Maybe DB.BlockId))